datacube/index/abstract.py

```diff
@@ -909,30 +909,38 @@ class AbstractIndex(ABC):
     inherit from this base class and implement all abstract methods.
     """
 
+    # Interface contracts
+    #   supports add() update() remove() etc methods.
+    supports_persistance = True
+    #   supports legacy ODCv1 EO style metadata types.
+    supports_legacy = True
+    #   supports non-geospatial (e.g. telemetry) metadata types
+    supports_nongeo = True
+
     @property
     @abstractmethod
     def url(self) -> str:
-        pass
+        ...
```

Comment:
My preferred style here is to include a docstring. Although, for a property it's of dubious utility.

Reply:
Good point, `AbstractIndex` and `AbstractIndexDriver` could both do with some inline documentation.